pkg/performanceprofile/controller/performanceprofile/resources/resources.go

```diff
@@ -27,20 +28,20 @@ func mergeMaps(src map[string]string, dst map[string]string) {
 
 // TODO: we should merge all create, get and delete methods
 
-func (r *PerformanceProfileReconciler) getMachineConfig(ctx context.Context, name string) (*mcov1.MachineConfig, error) {
+func GetMachineConfig(ctx context.Context, client client.Client, name string) (*mcov1.MachineConfig, error) {
```

Comment:
It could be worth adding a shallow object to hold a context and a client. When the client code makes repeated calls, it always use the same set of parameters.

Reply:
https://github.com/openshift/cluster-node-tuning-operator/pull/1032#discussion_r1569036144